dev-db/mycli/mycli-1.26.1-r1.ebuild

```diff
@@ -23,45 +23,61 @@ SRC_URI="
 LICENSE="BSD MIT"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="ssh"
 
 RDEPEND="
 	$(python_gen_cond_dep '
 		>=dev-python/cli_helpers-2.2.1[${PYTHON_USEDEP}]
 		>=dev-python/click-7.0[${PYTHON_USEDEP}]
-		>=dev-python/configobj-5.0.6[${PYTHON_USEDEP}]
+		>=dev-python/configobj-5.0.5[${PYTHON_USEDEP}]
 		>=dev-python/cryptography-1.0.0[${PYTHON_USEDEP}]
-		>=dev-python/prompt-toolkit-3.0.0[${PYTHON_USEDEP}]
+		>=dev-python/prompt-toolkit-3.0.6[${PYTHON_USEDEP}]
 		<dev-python/prompt-toolkit-4.0.0[${PYTHON_USEDEP}]
-		dev-python/pyaes[${PYTHON_USEDEP}]
-		>=dev-python/pygments-2.0[${PYTHON_USEDEP}]
+		>=dev-python/pyaes-1.6.1[${PYTHON_USEDEP}]
+		>=dev-python/pygments-1.6[${PYTHON_USEDEP}]
 		>=dev-python/pymysql-0.9.2[${PYTHON_USEDEP}]
-		dev-python/pyperclip[${PYTHON_USEDEP}]
+		>=dev-python/pyperclip-1.8.1[${PYTHON_USEDEP}]
+		>=dev-python/sqlglot-5.1.3[${PYTHON_USEDEP}]
 		>=dev-python/sqlparse-0.3.0[${PYTHON_USEDEP}]
-		<dev-python/sqlparse-0.5.0[${PYTHON_USEDEP}]
-		ssh? ( dev-python/paramiko[${PYTHON_USEDEP}] )'
+		<dev-python/sqlparse-0.5.0[${PYTHON_USEDEP}]'
 	)
 "
 BDEPEND="
 	test? (
 		$(python_gen_cond_dep '
-			dev-python/mock[${PYTHON_USEDEP}]
 			dev-python/paramiko[${PYTHON_USEDEP}]
 		')
 	)
 "
 
+EPYTEST_DESELECT=(
+	# Fails after a seemingly benign change in sqlparse 0.4.3
+	# https://github.com/dbcli/mycli/issues/1103
+	"test/test_smart_completion_public_schema_only.py::test_auto_escaped_col_names"
+	# Requires a running mysql daemon
+	"test/test_main.py::test_batch"
+	"test/test_main.py::test_execute"
+	"test/test_main.py::test_init"
+	"test/test_special_iocommands.py::test_favorite_query"
+	"test/test_special_iocommands.py::test_watch"
+	"test/test_tabular_output.py::test_sql_output"
+)
+
+EPYTEST_IGNORE=(
+	# Requires a running mysql daemon
+	"test/test_sqlexecute.py"
+)
+
 distutils_enable_tests pytest
 
 PATCHES=( "${FILESDIR}/mycli-1.21.1-fix-test-install.patch" )
 
-python_test() {
-	local EPYTEST_IGNORE=(
-		setup.py
-		mycli/magic.py
-		mycli/packages/parseutils.py
-		test/features
-		mycli/packages/paramiko_stub/__init__.py
-	)
-	epytest --capture=sys --doctest-modules --doctest-ignore-import-errors
+src_prepare() {
+	default
+
+	# Modify instruction to be Gentoo specific
```

Comment:
Since this is a pure Python package, having `IUSE=ssh` doesn't really hurt ("rebuilding" is just copying files after all) and it's generally cleaner to have users enable flags on packages than install random dependencies and have them clutter up `@world`.

Reply:
>  The usage of a USE flag should not control runtime dependencies when the package does not link to it. Doing so will create extra configuration for the package and re-compilation for no underlying file change on disk. This should be avoided and instead can be conveyed to the user via post install messages if needed. 

I'm getting mixed messages here. I was under the impression that dependencies like this should be avoided. Like the main reason for optfeature was this.

Is this is an area where there differing (but valid) schools of thought in in gentoo?

Reply:
We generally don't want USE flags if switching it would result in rebuilding big package involving lots of compilation that doesn't actually change what gets compiled ;-).